### PR TITLE
Make sure phoneNumber is not a required argument in the method

### PR DESCRIPTION
**Pull request by ashkan18**
# Problem
We get `ArgumentError (missing keyword: phone_number):` when we don't pass `phone_number` as an input argument to `setShipping` mutation, this arg is optional and is ok to not have it in input.

# Solution
Set the default to `nil` in the `resolve` method and add spec for checking that.

### CODE REVIEW

**Comment by ashkan18** [comment]
Sarah:
> merge on green

File: spec/controllers/api/requests/set_shipping_mutation_request_spec.rb [hunk 1]
```diff
@@ -73,7 +73,7 @@
             addressLine1: 'Vanak',
             addressLine2: 'P 80'
           }
-        }
+        }.compact
```
Comment (ds300) [inline spec/controllers/api/requests/set_shipping_mutation_request_spec.rb]: why is this needed?

Comment (ashkan18) [reply spec/controllers/api/requests/set_shipping_mutation_request_spec.rb]: in the original test we had, when setting `phoneNumber` to `nil` it was still passed as an argument. So we were testing `setShipping(phoneNumber: nil....)` the error was happening when someone was not passing `phoneNumber` so here `.compact` will remove `nil` params.
